volttron/platform/vip/agent/subsystems/pubsub.py

```diff
@@ -101,154 +104,41 @@ def __init__(self, core, rpc_subsys, peerlist_subsys, owner):
         self._peer_subscriptions = {}
         self._my_subscriptions = {}
         self.protected_topics = ProtectedPubSubTopics()
+        core.register('pubsub', self._handle_subsystem, self._handle_error)
+        self.vip_socket = None
+        self._results = ResultsDictionary()
+        self._event_queue = Queue()
+        self._retry_period = 300.0
+        self._processgreenlet = None
 
         def setup(sender, **kwargs):
             # pylint: disable=unused-argument
-            rpc_subsys.export(self._peer_sync, 'pubsub.sync')
-            rpc_subsys.export(self._peer_subscribe, 'pubsub.subscribe')
-            rpc_subsys.export(self._peer_unsubscribe, 'pubsub.unsubscribe')
-            rpc_subsys.export(self._peer_list, 'pubsub.list')
-            rpc_subsys.export(self._peer_publish, 'pubsub.publish')
-            rpc_subsys.export(self._peer_push, 'pubsub.push')
+            self._processgreenlet = gevent.spawn(self._process_loop)
             core.onconnected.connect(self._connected)
-            core.onviperror.connect(self._viperror)
             peerlist_subsys.onadd.connect(self._peer_add)
-            peerlist_subsys.ondrop.connect(self._peer_drop)
-
```

Comment:
We should remove all the stuff from the subsystem that you've disconnected here. 

All that should be left in the subsystem code is the client side of the pubsub. Everything left over from the old service agent needs to go.